src/proof/theory_proof.h

```diff
@@ -234,8 +258,12 @@ class TheoryProof {
    *
    * @param term expression representing term
    * @param os output stream
+   * @param expectedType The type that this is expected to have in a parent node.
+   *                      Null if there are no such requirements.
+   *                      This is useful for requesting type conversions from the theory.
+   *                      e.g. in (5.5 == 4) the right-hand-side should be converted to a real.
    */
-  virtual void printOwnedTerm(Expr term, std::ostream& os, const ProofLetMap& map) = 0;
+  virtual void printOwnedTerm(Expr term, std::ostream& os, const ProofLetMap& map, Type expectedType = Type()) = 0;
```

Comment:
I might be reading [this](https://github.com/CVC4/CVC4/wiki/Code-Style-Guidelines#default-arguments) wrong, but aren't you in one of the "banned" cases here?

Reply:
Good catch! You are correct. Andres and I cooked up an alternative. I'll implement it and update the PR.